### PR TITLE
Bug 2005 - Move filters to a common lib

### PR DESCRIPTION
**Pull request by jbuck**

### CODE REVIEW

File: cornfield/app.js [hunk 1]
```diff
@@ -341,19 +324,9 @@ app.get('/dashboard', function(req, res) {
   });
 });
 
-app.get('/api/projects', function(req, res) {
+app.get( '/api/projects', filter.isLoggedIn, filter.isStorageAvailable, function(req, res) {
```
Comment (dseif) [inline cornfield/app.js]: whitespace req, res

File: cornfield/app.js [hunk 2]
```diff
@@ -386,20 +359,10 @@ app.get('/api/projects', function(req, res) {
   });
 });
 
-app.get('/api/project/:id?', function(req, res) {
+app.get( '/api/project/:id?', filter.isLoggedIn, filter.isStorageAvailable, function(req, res) {
```
Comment (dseif) [inline cornfield/app.js]: same here

File: cornfield/app.js [hunk 3]
```diff
@@ -416,20 +379,10 @@ app.get('/api/project/:id?', function(req, res) {
   });
 });
 
-app.get('/api/delete/:id?', function(req, res) {
+app.get( '/api/delete/:id?', filter.isLoggedIn, filter.isStorageAvailable, function(req, res) {
```
Comment (dseif) [inline cornfield/app.js]: and hur

File: cornfield/lib/filter.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Options must be shared with anyone require()-ing this file
+var options;
+
+var filters = {
```
Comment (dseif) [inline cornfield/lib/filter.js]: make 2 var declarations into 1

File: cornfield/lib/filter.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  isLoggedIn: function( req, res, next ) {
+    if ( req.session.email ) {
+      next();
+    } else {
+      res.json( { error: 'unauthorized' }, 403 );
```
Comment (dseif) [inline cornfield/lib/filter.js]: this object literal and the one below should be multi-lined

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/de318bd880bb076/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/de318bd880bb076/output.txt

Total script time: 0.96 mins

Server is running on http://pr798.butyr.org/templates/basic/
Run the Test Suite: http://pr798.butyr.org/test/

File: cornfield/lib/filter.js (hunk 4 above)
Comment (jbuck) [inline cornfield/lib/filter.js]: I'm going to disagree with this... it makes it look ugly. And they're for separate purposes too! Plus, it's not like we're minifying the JS files, right?

Comment (dseif) [inline cornfield/lib/filter.js]:
```
var options,
    filters;

filters = {
```

That doesn't look too ugly to me. Multiple var declarations looks uglier.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/f361f7734edcc5f/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/f361f7734edcc5f/output.txt

Total script time: 1.12 mins

**Pull request merged**